services/catalog/rs_server_catalog/user_catalog.py

```diff
@@ -473,7 +478,41 @@ def manage_all_collections(self, collections: dict, auth_roles: list, user_login
         accessible_collections.extend(filter_collections(collections, user_login))
         return accessible_collections
 
-    async def manage_get_response(
+    def get_collection_id(self, collection: dict[str, str], size_owner_id: int) -> str:
+        """get the collection id with explicit typing
+
+        Args:
+            collection (dict[str, str]): The collection.
+            size_owner_id (int): The size of owner id.
+
+        Returns:
+            str: the collection id.
+        """
+        return collection["id"][size_owner_id:]
+
+    def update_links_for_all_collections(self, collections: list[dict]) -> list[dict]:
+        """Update the links for the endpoint /catalog/collections.
+
+        Args:
+            collections (list[dict]): all the collections to be updated.
+
+        Returns:
+            list[dict]: all the collections after the links updated.
+        """
+        for collection in collections:
+            owner_id = collection["owner"]
+            size_owner_id = int(
+                len(owner_id) + 1,
+            )  # example: if collection['id']=='toto_S1_L1' then size_owner_id=len('toto_')==len('toto')+1.
+            collection_id = self.get_collection_id(collection, size_owner_id)
+            # example: if collection['id']=='toto_S1_L1' then collection_id=='S1_L1'.
+            for i, link in enumerate(collection["links"]):
```

Comment:
for link in collection["links"]:
            link_parser = urlparse(link["href"])
            new_path = add_user_prefix(link_parser.path, owner_id, collection_id)
            link["href"] = link_parser._replace(path=new_path).geturl()